src/coreComponents/dataRepository/Group.cpp

```diff
@@ -156,6 +157,12 @@ void Group::processInputFileRecursive( xmlWrapper::xmlNode & targetNode )
       childName = childNode.name();
     }
 
+    // Make sure child names are not duplicated
+    GEOSX_ERROR_IF( std::find( childNames.begin(), childNames.end(), childName ) != childNames.end(),
+                    GEOSX_FMT( "Error: An XML block cannot contain children with duplicated names ({}/{}). ",
+                               getPath(), childName ) );
+    childNames.emplace_back( childName );
```

Comment:
I think this will error for unnamed nodes as well, since `childName` above is set to the tag name in that case?

Reply:
Agreed.  Also, in terms of merging named nodes, this is behavior that I've allowed in the xml preprocessor.  While it could lead to the same mistakes that we are trying to iron out, we do produce a compiled xml file that can be debugged